assets/js/lesson_6/array.js

```diff
@@ -0,0 +1,84 @@
+const arrInputMsg = 'Set the length of an array';
+const errorMsg = 'You must enter the number';
+const elementInputMsg = 'Put ellements in array';
+
+// Check if user entered valid value and set array length
+
+const mainArray = [];
+
+let arrayLengthInput = prompt(arrInputMsg);
+
+while (isNaN(arrayLengthInput) || arrayLengthInput <= 0) {
+    arrayLengthInput = prompt(`${arrInputMsg} \n
+${errorMsg}`)
+}
+
+/**
+ * Add elements to array.
+ * I could put 'counter' into 'for' cycle, but I'll use it in another variation 
+ * */
+
+let counter = arrayLengthInput;
+
+for (i = 1; i <= arrayLengthInput; i++) {
+    const elementInput = prompt(`${elementInputMsg} \n
+    Elements to add - ${counter--}`)
+
+    mainArray.push(elementInput)
+}
+
+console.log(mainArray)
+
+// Sort array, adding subarrays (numbers first)
+
+let numArray = []
+let strArray = []
+
+mainArray.forEach(a => {
+    if (Number(a)) {
+        numArray.push(Number(a))
+    } else {
+        strArray.push(a)
+    }
+})
+
+const sortedArray = numArray.slice().sort().concat(strArray.slice().sort())
```

Comment:
default `sort()` method will sort data like a `string`. Check the result with array, for example `[1, 2, 3, 12, 24, 41]`

Reply:
Didn't notice this thing. Thanks.
Resolved this with sort((a, b) => a - b)